jupyter_server/gateway/gateway_client.py

```diff
@@ -524,4 +604,9 @@ async def gateway_request(endpoint: str, **kwargs: ty.Any) -> HTTPResponse:
             f"appear to be valid.  Ensure gateway url is valid and the Gateway instance is running.",
         ) from e
 
+    cookie_values = response.headers.get("Set-Cookie")
+    if cookie_values:
+        cookie: SimpleCookie = SimpleCookie()
+        cookie.load(cookie_values)
+        GatewayClient.instance().update_cookies(cookie)
```

Comment:
I think it would be good to enclose this block within a check for whether the configuration is accepting cookies or not so we can skip the load when not enabled.
```suggestion
    if GatewayClient.instance().accept_cookies:
        cookie_values = response.headers.get("Set-Cookie")
        if cookie_values:
            cookie: SimpleCookie = SimpleCookie()
            cookie.load(cookie_values)
            GatewayClient.instance().update_cookies(cookie)
```

Reply:
Done.